src/utils/utils.service.js

```diff
@@ -59,6 +59,47 @@ class UtilsService {
       return index
     }, {})
   }
+
+  checkAndResolveConflicts (changedDoc, pouchdb) {
+    let preferredRevision = {}
+    if (!changedDoc._conflicts) {
+      return
+    }
+
+    if (!changedDoc.updatedAt) {
+      preferredRevision = changedDoc
+      changedDoc._conflicts.map((conflictingRev) => {
+        if (conflictingRev.updatedAt) {
+          preferredRevision = conflictingRev
+        } else {
+          pouchdb.remove(changedDoc._id, conflictingRev._rev)
+        }
+      })
+    } else {
+      let serializedRevisions = this.serialiseDocWithConflictsByProp(changedDoc, 'updatedAt')
+      preferredRevision = serializedRevisions.pop()
+      serializedRevisions.forEach(revision => {
+        pouchdb.remove(changedDoc._id, revision)
+      })
+    }
+    pouchdb.put(preferredRevision) // do we still need to PUT after plucking conflicts?
+  }
+
+  serialiseDocWithConflicts (Doc, prop) {
+    let mainDoc = {}
+    let serialisedDocs = []
+    Object.keys(Doc).map((prop) => {
```

Comment:
Why iterate over all keys if you're only looking for `_conflicts`? Can't you do `if (Doc._conflicts) mainDoc._conflicts = Doc._conflicts`?